File: oim-jruby/create_user.rb
require 'java'
require 'xlclient'

include_class('java.lang.Exception') {|package,name| "J#{name}" }
include_class 'java.lang.System' 
include_class 'java.lang.Thread' 
include_class 'java.util.HashMap'
include_class 'java.util.Hashtable'
include_class('Thor.API.tcUtilityFactory') {|package,name| "OIM#{name}"}


login = ARGV[0]

xlclient = XLAPIClient.new
#xlclient.defaultLogin
#jndi = Hashtable.new({
#    'java.naming.provider.url' => 'ormi://dadvmn0695.us.oracle.com:23791/Xellerate',
#    'java.naming.factory.initial' => 'oracle.j2ee.rmi.RMIInitialContextFactory'
#})


#xlclient.defaultLogin
xlclient.passwordLogin('bigadmin', 'foo')
#xlclient.signatureLogin('xelsysadm')
#xlclient.remoteLogin('xelsysadm', 'xelsysadm', jndi)

usrIntf = xlclient.getUtility('usr')

usrHash = {
  'Users.User ID' => login ,
  'Users.First Name' => login + '_First' ,
  'Users.Last Name' => login + '_Last' ,
  'Users.Middle Name' => login + '_Middle' ,
  'Users.Password' => 'foo',
  'Users.Role' => 'Full-Time',
  'Users.Xellerate Type' => 'End-User',
  'Organizations.Key' => '1'
  #'USR_UDF_CWFUDF1' => 'hello',
  #'USR_UDF_CWFUDF2' => 'world'
}

usrMap = HashMap.new(usrHash)

t1 = System.currentTimeMillis
usrKey = usrIntf.createUser(usrMap)
t2 = System.currentTimeMillis

delta = t2-t1
puts "Created user with key = #{usrKey} time = #{delta}"

xlclient.close
System.exit 0

File: oim-scala/create_user.scala
import scala.collection.mutable._
import java.util.HashMap
import java.lang.Long

import Thor.API.Operations._

import OIM._

object CreateUser {
    def main(args: Array[String]) {

        val login: String = args(0)

        // create a user
        val factory = getFactory("xelsysadm", "xelsysadm")
        val usrIntf = getInterface(factory, "usr").asInstanceOf[tcUserOperationsIntf]

        val map = Map[String,Object](
            "Users.User ID" -> login,
            "Users.First Name" -> (login + "First"),
            "Users.Middle Name" -> (login + "Middle"),
            "Users.Last Name" -> (login + "Last"),
            "Users.Password" -> "foo",
            "Users.Role" -> "Full-Time",
            "Users.Xellerate Type" -> "End-User",
            "Organizations.Key" -> "1",
            "Users.Email" -> (login + "@oracle.com")
        )

        val usrMap = getJavaMap(map)
        val usrKey = usrIntf.createUser(usrMap)

        println("User created login = " + login + "  key = " + usrKey)

        factory.close()
    }
}
